From 6c8c7dae421abb0481d9eb17b4a4d819024258c4 Mon Sep 17 00:00:00 2001
From: Florent Revest <revestflo@gmail.com>
Date: Fri, 19 Jul 2019 19:36:42 +0200
Subject: [PATCH 3/3] hci_core: handle incorrect specified max page number

Some chips seem to report a different maximum page number than they
allow us to read. Therfor we check for the status code after each
read and skip those reporting out-of-range reads.
Upstream-Status: Inappropriate
---
 net/bluetooth/hci_event.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/net/bluetooth/hci_event.c b/net/bluetooth/hci_event.c
index d57c11c1c6b5..ea15551fbf84 100644
--- a/net/bluetooth/hci_event.c
+++ b/net/bluetooth/hci_event.c
@@ -2867,6 +2867,14 @@ static void hci_cmd_complete_evt(struct hci_dev *hdev, struct sk_buff *skb,
 
 	case HCI_OP_READ_LOCAL_EXT_FEATURES:
 		hci_cc_read_local_ext_features(hdev, skb);
+
+ 		if (*status == 0x30) {
+			/* If we end up here the BT chip might have report a different maximum
+			 * page number than it allows us to read when we try. As this would
+			 * break our initialization process but isn't really preventing us
+			 * from finishing we're switching the status back to a sane one. */
+			*status = 0x0;
+		}
 		break;
 
 	case HCI_OP_READ_BUFFER_SIZE:
-- 
2.22.0

